### PR TITLE
[AMS-1554] Migrate to .NET Core 2.2

### PR DESCRIPTION
**Pull request by xakep139**

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -68,8 +68,9 @@ build:vstore-host-image:
   when: manual
   allow_failure: false
   script:
-    - TAG=$CI_TAG DOCKER_FILE=publish/vstore/Dockerfile DOCKER_BUILD_CONTEXT=publish/vstore make docker-build
-    - TAG=$CI_TAG make docker-push
```
Comment (xakep139) [inline .gitlab-ci.yml]: Do not use CI starter kit

**Comment by xakep139** [comment]
We now cannot publish with `alpine.3.8-x64` RID, because of dotnet/corefx#34316

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -9,22 +9,22 @@ stages:
...
   script:
-    - dotnet restore --runtime ubuntu.16.04-x64
-    - dotnet test tests/VStore.UnitTests/VStore.UnitTests.csproj --configuration Release /p:CollectCoverage=true /p:CoverletOutputFormat=lcov /p:CoverletOutput="`pwd`/coverage/"
-    - dotnet publish src/VStore.Host --configuration Release --runtime ubuntu.16.04-x64 --output ../../publish/vstore
-    - dotnet publish src/VStore.Worker --configuration Release --runtime ubuntu.16.04-x64 --output ../../publish/vstore-worker
-    - dotnet restore --runtime alpine.3.7-x64
-    - dotnet publish src/VStore.Renderer --configuration Release --runtime alpine.3.7-x64 --output ../../publish/vstore-renderer
+    - dotnet restore --runtime alpine.3.9-x64
+    - dotnet test tests/VStore.UnitTests/VStore.UnitTests.csproj --configuration Release /p:CollectCoverage=true /p:Include="[Build]*" /p:CoverletOutputFormat=lcov /p:CoverletOutput="`pwd`/coverage/"
+    - dotnet publish src/VStore.Host --configuration Release --runtime alpine.3.9-x64 --output ../../publish/vstore
```
Comment (xakep139) [inline .gitlab-ci.yml]: Build is failing here due to https://github.com/dotnet/sdk/issues/3037

File: src/VStore.Worker/Jobs/BinariesCleanupJob.cs [hunk 3]
```diff
@@ -124,7 +124,7 @@ private static RetryPolicy CreateRetryPolicy(ILogger logger)
         {
             return Policy<List<KafkaEvent<BinaryReferencedEvent>>>
                 .HandleResult(result => result.Count == 0)
-                .WaitAndRetryForeverAsync(
+                .WaitAndRetryForever(
```
Comment (igor-moskvitin) [inline src/VStore.Worker/Jobs/BinariesCleanupJob.cs]: Why not `WaitAndRetryForeverAsync`?

Comment (xakep139) [reply src/VStore.Worker/Jobs/BinariesCleanupJob.cs]: Some Kafka APIs is now sync

**Pull request merged**